exercises/practice/two-bucket/.meta/examples/success-standard/src/TwoBucket.hs

```diff
@@ -0,0 +1,85 @@
+module TwoBucket (measure) where
+
+import Control.Monad.State ( evalState, MonadState(put, get), StateT )
+import qualified Control.Monad.Identity as Data.Functor.Identity
+import Data.Maybe (mapMaybe)
+
+-- Current status of the buckets, or their initial capacity
+type Buckets = (Int, Int)
+-- Sequence of moves
+type Moves = [Buckets]
+-- Solution candidates
+type Candidates = [Moves]
+
+measure :: Buckets -> Int -> (Int, Buckets)
+measure capacity target =
+    if null solution
+    then (0, (0, 0))
```

Comment:
Why not use `Maybe` and return `Nothing` here?

Reply:
Changed.